src/v/cluster/controller_stm.h

```diff
@@ -32,7 +31,6 @@ class controller_stm final
       topic_updates_dispatcher,
       security_manager,
       members_manager,
-      data_policy_manager,
```

Comment:
if a log contains an old data policy batch, then during replay an applicable state machine won't be found and the vassert for `!state` condition in mux_state_machine::apply will fail, right? i was assuming that the reason the adl<> stuff was left in was for this case of handling old data policy batches, but now i'm not sure.

Reply:
you are right, i will add a no op implemenation to consume those data policy batches